src/Runner/Parallel/Process.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use PhpCsFixer\Console\Command\FixCommand;
+use PhpCsFixer\Runner\RunnerConfig;
+use PhpCsFixer\ToolInfo;
+use React\ChildProcess\Process as ReactProcess;
+use React\EventLoop\LoopInterface;
+use React\EventLoop\TimerInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\WritableStreamInterface;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\ArgvInput;
+
+/**
+ * Represents single process that is handled within parallel run.
+ * Inspired by:
+ *   - https://github.com/phpstan/phpstan-src/blob/9ce425bca5337039fb52c0acf96a20a2b8ace490/src/Parallel/Process.php
+ *   - https://github.com/phpstan/phpstan-src/blob/1477e752b4b5893f323b6d2c43591e68b3d85003/src/Process/ProcessHelper.php.
+ *
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class Process
+{
+    // Properties required for process instantiation
+    private string $command;
+    private LoopInterface $loop;
+    private int $timeoutSeconds;
+
+    // Properties required for process execution
+    private ReactProcess $process;
+    private ?WritableStreamInterface $in = null;
+
+    /** @var false|resource */
+    private $stdErr;
+
+    /** @var false|resource */
+    private $stdOut;
+
+    /** @var callable(mixed[]): void */
+    private $onData;
+
+    /** @var callable(\Throwable): void */
+    private $onError;
+
+    private ?TimerInterface $timer = null;
+
+    public function __construct(string $command, LoopInterface $loop, int $timeoutSeconds)
+    {
+        $this->command = $command;
+        $this->loop = $loop;
+        $this->timeoutSeconds = $timeoutSeconds;
+    }
+
+    public static function create(
+        LoopInterface $loop,
+        RunnerConfig $runnerConfig,
+        ProcessIdentifier $identifier,
+        int $serverPort
+    ): self {
+        $input = self::getArgvInput();
+
+        $commandArgs = [
+            'php',
```

Comment:
Maybe you can use `PHP_BINARY` and fallback to `php` when not defined. https://www.php.net/manual/en/reserved.constants.php

Reply:
Symfony has a helper for finding the php executable and spawning it again. Right ? Is there any advantage to using that instead of this? (Assuming symfony console / process is already used in the code )

Reply:
I was talking about:
https://symfony.com/doc/current/components/process.html#executing-a-php-child-process-with-the-same-configuration
and
https://symfony.com/doc/current/components/process.html#finding-the-executable-php-binary

I don't know if it's related to what you use it for, my brain just went 💡 and maybe this is the better way or easier way. If it's not what you want, then ignore it :).

Reply:
(I see ProcessLinter uses the `PhpExecutableFinder`, but Parallel/Process.php doesn't, from looking at the code snippet above?)

Reply:
I think the initial comment of this thread is pretty important, to share a use case: have a php binary somewhere on the computer but not as the default php executable. In this case there is no default php executable.
So you do /path/to/php8.2  /bin/composer run phpcs

And then in composer.json you have a scripts section:
```json
"phpcs": "@php ./vendor/bin/phpcs"
```
This special syntax will keep using the correct php binary 
it is different from 
```json
"phpcs": "./vendor/bin/phpcs"
```
where this would use the default php that might not be the one running composer binary. 

Anyway, always use the same php version as the executor when running another php process.

Reply:
Thanks @shyim for spotting this, it's improved now.